readthedocs/vcs_support/backends/git.py

```diff
@@ -151,27 +156,20 @@ def parse_branches(self, data):
               origin/release/2.0.0
               origin/release/2.1.0
         """
-        clean_branches = []
-        # StringIO below is expecting Unicode data, so ensure that it gets it.
-        if not isinstance(data, str):
-            data = str(data)
-        raw_branches = csv.reader(StringIO(data), delimiter=' ')
-        for branch in raw_branches:
-            branch = [f for f in branch if f != '' and f != '*']
-            # Handle empty branches
-            if branch:
-                branch = branch[0]
-                if branch.startswith('origin/'):
-                    cut_len = len('origin/')
-                    slug = branch[cut_len:].replace('/', '-')
-                    if slug in ['HEAD']:
-                        continue
-                    clean_branches.append(VCSVersion(self, branch, slug))
-                else:
-                    # Believe this is dead code.
-                    slug = branch.replace('/', '-')
-                    clean_branches.append(VCSVersion(self, branch, slug))
-        return clean_branches
+        branches = []
+        for match in self.BRANCH_REGEX.finditer(data):
```

Comment:
I feel like we're depending on some regex magic here that isn't easy to understand. It would be good to document exactly how this is working. Are we still handling empty branches, and `*` names? Are they getting filtered out with regex magic?